From 36d6772a76adeb0a55425ca9ec30908f8fe8509c Mon Sep 17 00:00:00 2001
From: Alan Wu <XrXr@users.noreply.github.com>
Date: Fri, 25 Oct 2019 18:31:13 -0400
Subject: [PATCH] Set mtime to avoid fakiness

GZIP file format has a timestamp in its header, which makes it so that
making a GZIP file with the default setting is only deterministic if
done during the same second. Set mtime to 1 instead of taking the
current time in the header. 0 does not work as that indicates to zlib
to take the current time.
---
 lib/cacheable.rb | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/cacheable.rb b/lib/cacheable.rb
index eed42a8..1a98639 100644
--- a/lib/cacheable.rb
+++ b/lib/cacheable.rb
@@ -32,6 +32,7 @@ module Cacheable
   def self.compress(content)
     io = StringIO.new
     gz = Zlib::GzipWriter.new(io)
+    gz.mtime = 1
     gz.write(content)
     io.string
   ensure
-- 
2.19.1

